src/sound/source.hpp

```diff
@@ -0,0 +1,159 @@
+#pragma once
+
+#include "external.hpp"
+#include "clip.hpp"
+
+class SoundGroup;
+#include "group.hpp"
+
+class SoundSourceObject {
+private:
+
+	/// Source
+	ALuint source;
+	
+	/// Shared pointer to SoundGroup where we get offset for parameters and movement
+	std::shared_ptr<SoundGroup> sso_sg;
+	/// Weak pointer to SoundClip where we get buffer
+	std::weak_ptr<SoundClip> sc_buffer;

```

Comment:
`weak_ptr`? 
I this this should be `shared_ptr` to avoid the buffer being deleted while in use